### PR TITLE
[STY] Fix ruff errors RUF002-3 and RUF005

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4579

Changes proposed in this pull request:

- Fix ruff errors RUF002-3 and RUF005

### CODE REVIEW

File: nilearn/datasets/tests/test_utils.py [hunk 1]
```diff
@@ -545,7 +545,7 @@ def test_fetch_files_overwrite(
     fil = _utils.fetch_files(
         data_dir=str(tmp_path),
         verbose=0,
-        files=[files + ({"overwrite": True},)],
+        files=[(*files, {"overwrite": True})],
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_utils.py:548]:
```suggestion
        files=[*files, {"overwrite": True}],
```

**Comment by Remi-Gau** [comment]
Seems ruff really like the "splat" operator

**Review by bthirion — APPROVED** [review]
Not particularly enthusiastic about the change set but LGTM.

**Comment by Remi-Gau** [comment]
> Not particularly enthusiastic about the change

yeah this is partly subjective but I have been confused and/or burned so many times by a trailing coma turning things into a tuple that I kind of prefer things that can avoid it.

**Review by man-shu — APPROVED** [review]
LGTM!

**Pull request merged**